src/libstd/sync/mutex.rs

```diff
@@ -243,6 +245,52 @@ impl<T: ?Sized> Mutex<T> {
     pub fn is_poisoned(&self) -> bool {
         self.inner.poison.get()
     }
+
+    /// Consumes this mutex, returning the underlying data.
+    ///
+    /// # Failure
+    ///
+    /// If another user of this mutex panicked while holding the mutex, then
+    /// this call will return an error instead.
+    #[inline]
```

Comment:
Ah as one final piece, could these `#[inline]` annotations be left out? These are pretty hefty functions and the inline shouldn't be needed anyway as a result of the generic parameter on `Mutex`